xclim/core/units.py

```diff
@@ -463,6 +463,25 @@ def infer_sampling_units(
     return out
 
 
+DELTA_ABSOLUTE_TEMP = {
+    units.delta_degC: units.kelvin,
+    units.delta_degF: units.rankine,
+}
+
+
+def ensure_absolute_temperature(units: str):
+    """Convert temperature units to their absolute counterpart, assuming they represented a difference (delta).
+
+    Celsius becomes Kelvin, Fahrenheit becomes Rankine. Does nothing for other units.
+    """
+    a = str2pint(units)
+    # ensure a delta pint unit
+    a = a - 0 * a
```

Comment:
I'm sorry, but this is no always `a` ?

Reply:
Yes and no. The magnitude is always exactly the same, but the units can be different. 

Pint handles non-multiplicative (or _non-absolute_, or _offset_) units differently. A temperature substraction cannot return a temperature, instead is returns a "delta temperature" in pint. Ex: `1 degree_Celsius - 0 degree_Celsius = 1 delta_degree_celsius`.  This PR takes the concept one step further by converting the "delta temperature" in its absolute equivalent, since CF does not recognize delta temperatures.

See https://pint.readthedocs.io/en/stable/user/nonmult.html

Reply:
ah ok I see... thanks for the explanation :-)